pkg/virtctl/spice/spice.go

```diff
@@ -0,0 +1,102 @@
+package spice
+
+import (
+	"errors"
+	"fmt"
+	flag "github.com/spf13/pflag"
+	"io/ioutil"
+	"k8s.io/client-go/tools/clientcmd"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+const FLAG = "spice"
+const SPICE_FILENAME = "console.vv"
+const VM_SPICE_URL = "http://%s/apis/kubevirt.io/v1alpha1/namespaces/default/vms/%s/spice"
+
+type Spice struct {
+}
+
+func DownloadSpice(vm, host string) (string, error) {
+	var path string = fmt.Sprintf(VM_SPICE_URL, host, vm)
+	req, err := http.NewRequest("GET", path, nil)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Can't create request: %s", err.Error()))
+	}
+	req.Header.Add("Accept", "text/plain")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		if resp != nil && resp.StatusCode != http.StatusOK {
+			return "", errors.New(fmt.Sprintf("Can't connect to console (%d): %s\n", resp.StatusCode, err.Error()))
+		}
+		return "", errors.New(fmt.Sprintf("Can't connect to console: %s\n", err.Error()))
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Can't read body: %s\n", err.Error()))
+	}
+	return fmt.Sprintf("%s", body), nil
+}
+
+func (o *Spice) FlagSet() *flag.FlagSet {
+
+	cf := flag.NewFlagSet(FLAG, flag.ExitOnError)
+	cf.BoolP("details", "d", false, "If present, print SPICE console to stdout, otherwise run remote-viewer")
+	return cf
+}
+
+func (o *Spice) Run(flags *flag.FlagSet) int {
+	server, _ := flags.GetString("server")
+	kubeconfig, _ := flags.GetString("kubeconfig")
+	details, _ := flags.GetBool("details")
+
+	if len(flags.Args()) != 2 {
+		log.Println("VM name is missing")
+		return 1
+	}
+	vm := flags.Arg(1)
+
+	config, err := clientcmd.BuildConfigFromFlags(server, kubeconfig)
+
+	if err != nil {
+		log.Println(err)
+		return 1
+	}
+	body, err := DownloadSpice(vm, config.Host)
+	if err != nil {
+		log.Fatalf(err.Error())
+		return 1
+	}
+	if details {
+		fmt.Printf("%s", body)
+	} else {
+		f, err := os.Create(SPICE_FILENAME)
+
+		if err != nil {
+			log.Fatalf("Can't open file: %s", err.Error())
+			return 1
+		}
+		defer f.Close()
+
+		_, err = f.WriteString(body)
+		if err != nil {
+			log.Fatalf("Can't write to file: %s", err.Error())
+			return 1
+		}
+
+		f.Sync()
+
+		cmnd := exec.Command("remote-viewer", "console.vv")
+		cmnd.Start()
+	}
+	return 0
+}
+
+func (o *Spice) Usage() string {
+	usage := "The following options can be passed to any command:\n\n"
```

Comment:
Here some usage examples like in  `virtctl/console/console.go` would be nice here.